From 7027ea1b867ea342ae209bef6405d71c6ffd4283 Mon Sep 17 00:00:00 2001
From: Yonggang Luo <luoyonggang@gmail.com>
Date: Fri, 3 Dec 2021 03:14:45 +0800
Subject: [PATCH 6/6] win32: The opengl headers should install to mesa/GL
 subdir

The subdir `mesa/GL` used to avoid conflict between Windows GL headers and mesa GL headers.
The GL headers resident in Windows SDK are:
```
gl.h
glaux.h
glcorearb.h
glext.h
glu.h
glxext.h
wgl.h
wglext.h
```

Signed-off-by: Yonggang Luo <luoyonggang@gmail.com>
---
 include/meson.build | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/meson.build b/include/meson.build
index d1d5787cf1d..f4686cd3eb0 100644
--- a/include/meson.build
+++ b/include/meson.build
@@ -62,11 +62,15 @@ if not with_glvnd
   endif
 
   if with_opengl
+    opengl_subdir = 'GL'
+    if host_machine.system() == 'windows'
+      opengl_subdir = 'mesa/GL'
+    endif
     install_headers(
       'GL/gl.h',
       'GL/glcorearb.h',
       'GL/glext.h',
-      subdir : 'GL',
+      subdir : opengl_subdir,
     )
   endif
 
-- 
2.31.1.windows.1

